src/vs-bicep/Bicep.VSLanguageServerClient/DidChangeWatchedFilesNotifier.cs

```diff
@@ -0,0 +1,135 @@
+
+                    if (workspaceServices is not null && workspaceServices.Any())
+                    {
+                        var workspace = workspaceServices.First();
+                        var currentWorkspace = workspace.Value.CurrentWorkspace;
+
+                        if (currentWorkspace is not null)
+                        {
+                            location = currentWorkspace.Location;
+                        }
+                    }
+                }
+            }
+        }
+
+        public void CreateFileSystemWatchers()
+        {
+            if (rpc is not null && location is not null)
+            {
+                foreach (string filter in filters)
+                {
+                    var fileSystemWatcher = new FileSystemWatcher(location);
+
+                    fileSystemWatcher.NotifyFilter = NotifyFilters.Attributes
+                         | NotifyFilters.CreationTime
+                         | NotifyFilters.DirectoryName
+                         | NotifyFilters.FileName
+                         | NotifyFilters.LastAccess
+                         | NotifyFilters.LastWrite
+                         | NotifyFilters.Security
+                         | NotifyFilters.Size;
+                    fileSystemWatcher.IncludeSubdirectories = true;
+                    fileSystemWatcher.EnableRaisingEvents = true;
+                    fileSystemWatcher.Filter = filter;
+                    fileSystemWatcher.Changed += OnChanged;
+                    fileSystemWatcher.Created += OnCreated;
+                    fileSystemWatcher.Deleted += OnDeleted;
+
+                    fileSystemWatchers.Add(fileSystemWatcher);
+                }
+            }
+        }
+
+        private void OnChanged(object sender, FileSystemEventArgs e) =>
+            SendWorkspaceDidChangeConfigurationNotificationAsync(FileChangeType.Changed, e.FullPath);
+
+        private void OnCreated(object sender, FileSystemEventArgs e) =>
+            SendWorkspaceDidChangeConfigurationNotificationAsync(FileChangeType.Created, e.FullPath);
+
+        private void OnDeleted(object sender, FileSystemEventArgs e) =>
+            SendWorkspaceDidChangeConfigurationNotificationAsync(FileChangeType.Deleted, e.FullPath);
+
+        private Task SendWorkspaceDidChangeConfigurationNotificationAsync(FileChangeType fileChangeType, string fullPath)
+        {
+            var fileEvent = new FileEvent()
+            {
+                FileChangeType = fileChangeType,
+                Uri = new Uri(fullPath)
+
+            };
+            var didChangeWatchedFilesParams = new DidChangeWatchedFilesParams()
+            {
+                Changes = new FileEvent[] { fileEvent }
+            };
+
+            _ = rpc.NotifyWithParameterObjectAsync(Methods.WorkspaceDidChangeWatchedFilesName, didChangeWatchedFilesParams);
+
+            return Task.CompletedTask;
+        }
+
+        public void Dispose()
+        {
+            if (fileSystemWatchers.Any())
```

Comment:
Should you not guard against multiple Dispose() calls?  It's been a while, but I thought the dispose pattern required that.

Reply:
Hmmm I am not sure. I saw some example usages and didn't see anything specific. Let me know if you think a lock is required?

Reply:
How about clearing fileSystemWatchers?